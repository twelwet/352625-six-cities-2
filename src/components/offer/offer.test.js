import React from "react";
import renderer from "react-test-renderer";
import Offer from "../offer/offer";
import {BrowserRouter} from "react-router-dom";

const onOfferHover = () => {};
const onOfferLeave = () => {};
const onBookmarkClick = () => {};

const offer = {
  [`id`]: 111,
  [`city`]: {
    [`name`]: `City Name`,
    [`location`]: {
      [`latitude`]: 52.38333,
      [`longitude`]: 4.9,
      [`zoom`]: 12
    }
  },
  [`location`]: {
    [`latitude`]: 52.3909553943508,
    [`longitude`]: 4.85309666406198,
    [`zoom`]: 12
  },
  [`title`]: `Some text`,
  [`type`]: `Apartment`,
  [`price`]: 120,
  [`preview_image`]: `img/apartment-01.jpg`,
  [`rating`]: 100,
  [`is_premium`]: true,
  [`is_favorite`]: false
};

it(`Offer correctly renders after relaunch`, () => {
  const tree = renderer
    .create(
        <BrowserRouter>
          <Offer
            id={offer[`id`]}
            key={offer[`id`]}
            title={offer[`title`]}
            type={offer[`type`]}
            price={offer[`price`]}
            preview_image={offer[`preview_image`]}
            rating={offer[`rating`]}
            is_premium={offer[`is_premium`]}
            is_favorite={offer[`is_favorite`]}
            onOfferHover={onOfferHover}
            onOfferLeave={onOfferLeave}
            onBookmarkClick={onBookmarkClick}
          />
        </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
