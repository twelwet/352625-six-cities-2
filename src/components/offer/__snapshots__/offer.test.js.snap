// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offer correctly renders after relaunch 1`] = `
<div
  className="place-card__info"
>
  <div
    className="place-card__price-wrapper"
  >
    <div
      className="place-card__price"
    >
      <b
        className="place-card__price-value"
      >
        €
        120
      </b>
      <span
        className="place-card__price-text"
      >
        / night
      </span>
    </div>
    <button
      className="place-card__bookmark-button button"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="place-card__bookmark-icon"
        height="19"
        width="18"
      >
        <use
          xlinkHref="#icon-bookmark"
        />
      </svg>
      <span
        className="visually-hidden"
      >
        To bookmarks
      </span>
    </button>
  </div>
  <div
    className="place-card__rating rating"
  >
    <div
      className="place-card__stars rating__stars"
    >
      <span
        style={
          Object {
            "width": "2000%",
          }
        }
      />
      <span
        className="visually-hidden"
      >
        Rating
      </span>
    </div>
  </div>
  <h2
    className="place-card__name"
  >
    <a
      href="/offer/111"
      onClick={[Function]}
    >
      Some text
    </a>
  </h2>
  <p
    className="place-card__type"
  >
    Apartment
  </p>
</div>
`;
